tests/integration/commands/clone/clone.test.ts

```diff
@@ -0,0 +1,421 @@
+import { describe, expect, it, beforeEach, afterEach } from 'vitest'
+import js from 'dedent'
+
+import { withMockApi } from '../../utils/mock-api.js'
+import { withSiteBuilder } from '../../utils/site-builder.js'
+import { answerWithValue, CONFIRM, DOWN, handleQuestions } from '../../utils/handle-questions.js'
+import { createMock } from '../../utils/mock-execa.js'
+import execa from 'execa'
+import { cliPath } from '../../utils/cli-path.js'
+
+describe('clone command', () => {
+  let execaMock: Awaited<ReturnType<typeof createMock>>[0]
+  let removeExecaMock: Awaited<ReturnType<typeof createMock>>[1]
+
+  beforeEach(async () => {
+    ;[execaMock, removeExecaMock] = await createMock(js`
```

Comment:
oof, this took me a while to figure out. Because we run CLI commands via a spawned child process, we can't actually mock anything. This seemingly hasn't come up much before, but here we're interacting with a third-party server (to clone from git remotes), so it's strongly desirable to mock that out. It turns out there was already this helper and pattern for this. Hope this is clear enough...